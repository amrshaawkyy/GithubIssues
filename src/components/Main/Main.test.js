import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Main from './Main';
import ButtonMain from '../UI/ButtonMain/ButtonMain';

configure({ adapter: new Adapter() });


describe('<Main />', () => {
    const wrapper = shallow(<Main />);
    afterEach(() => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render one <ButtonMain /> element', () => {
        expect(wrapper.find(ButtonMain)).toHaveLength(1);
    });
    it('renders text', () => {
        const text = "Issues";
        const renderedComponent = shallow(
            <ButtonMain>{text}</ButtonMain>
        );
        expect(
            renderedComponent.contains(text)
        ).toEqual(true);
    });
});
