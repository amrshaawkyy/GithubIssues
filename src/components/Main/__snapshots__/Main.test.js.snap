// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders text 1`] = `ShallowWrapper {}`;

exports[`<Main /> should render one <ButtonMain /> element 1`] = `ShallowWrapper {}`;
